src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -159,20 +155,21 @@ public static Tensor<T> ConcatenateOnDimension<T>(int dimension, params scoped R
                     }
                     sumOfAxis += tensors[i].Lengths[dimension];
                 }
-            }
 
-            Tensor<T> tensor;
-            if (dimension == -1)
-            {
-                tensor = Tensor.Create<T>([totalLength]);
-            }
-            else
-            {
                 nint[] lengths = new nint[tensors[0].Rank];
                 tensors[0].Lengths.CopyTo(lengths);
                 lengths[dimension] = sumOfAxis;
                 tensor = Tensor.Create<T>(lengths);
             }
+            else
+            {
+                // Calculate total space needed.
+                nint totalLength = 0;
+                for (int i = 0; i < tensors.Length; i++)
```

Comment:
Same here, does this need to check for overflow?

Reply:
Here it wouldn't be a bad idea to add a check.

Reply:
Actually here we are creating a new tensor, and the backing array can't have more than int.max elements right? So really I should be making `totalLength` an int and then checking for overflow.

Reply:
You're adding `flattenedLength` together, so the risk is that `flattenedLength` overflows.

Reply:
This can happen simply combining 16x `Tensor<byte>` that have a flattened length of `134_217_728` (128MB of data) on a 32-bit machine.